drivers/storage/partmgr/partition.c

```diff
@@ -173,6 +172,103 @@ PartitionHandleStartDevice(
     return STATUS_SUCCESS;
 }
 
+/**
+ * @brief
+ * Notifies MountMgr to delete all mount points
+ * associated with the given volume.
+ *
+ * @note    This should belong to volmgr.sys and act on a PVOLUME_EXTENSION.
+ **/
+static
+CODE_SEG("PAGE")
+NTSTATUS
+VolumeDeleteMountPoints(
+    _In_ PPARTITION_EXTENSION PartExt)
+{
+    NTSTATUS Status;
+    UNICODE_STRING MountMgr;
+    ULONG InputSize, OutputSize;
+    LOGICAL Retry;
+    PUNICODE_STRING DeviceName;
+    PDEVICE_OBJECT DeviceObject;
+    PFILE_OBJECT FileObject = NULL;
+    PMOUNTMGR_MOUNT_POINT InputBuffer = NULL;
+    PMOUNTMGR_MOUNT_POINTS OutputBuffer = NULL;
+
+    PAGED_CODE();
+
+    /* Get the device pointer to the MountMgr */
+    RtlInitUnicodeString(&MountMgr, MOUNTMGR_DEVICE_NAME);
+    Status = IoGetDeviceObjectPointer(&MountMgr,
+                                      FILE_READ_ATTRIBUTES,
+                                      &FileObject,
+                                      &DeviceObject);
+    if (!NT_SUCCESS(Status))
+        return Status;
+
+    /* Setup the volume device name for deleting its mount points */
+    DeviceName = &PartExt->DeviceName;
+
+    /* Allocate the input buffer */
+    InputSize = sizeof(MOUNTMGR_MOUNT_POINT) + DeviceName->Length;
+    InputBuffer = ExAllocatePoolZero(PagedPool, InputSize, TAG_PARTMGR);
+    if (!InputBuffer)
+    {
+        Status = STATUS_INSUFFICIENT_RESOURCES;
+        goto Quit;
+    }
+
+    /* Fill it in */
+    InputBuffer->DeviceNameOffset = sizeof(MOUNTMGR_MOUNT_POINT);
+    InputBuffer->DeviceNameLength = DeviceName->Length;
+    RtlCopyMemory(&InputBuffer[1], DeviceName->Buffer, DeviceName->Length);
+
+    /*
+     * Start with a guessed large-enough buffer size to begin with, then
+     * call IOCTL_MOUNTMGR_DELETE_POINTS. If the buffer is not big enough,
+     * use the value obtained from MOUNTMGR_MOUNT_POINTS::Size to re-allocate
+     * a suitable buffer and call the IOCTL once more.
+     */
+    OutputSize = PAGE_SIZE; // Must be >= sizeof(MOUNTMGR_MOUNT_POINTS)
```

Comment:
```suggestion
    C_ASSERT(PAGE_SIZE >= sizeof(MOUNTMGR_MOUNT_POINTS));
    OutputSize = PAGE_SIZE;
```

Reply:
I will instead put the comment in the comment block above. The "must be >= ...." was a generic comment for what the OutputSize has to satisfy.